Select_GCMs_Centroid.Rmd

```diff
@@ -379,18 +383,151 @@ dualscatter  + geom_text_repel(aes(label=GCM)) +
 ```
 
-For selected models, look at change in temp and precip over time:
+Presentation Figure:
+
+```{r}
+###Scatter plot showing delta precip and tavg, color by emissions scenario, with box for central CF
+
+ggplot(
+  future_means,
+  aes(
+    delta_tavg,
+    365 * delta_pr,
+    xmin = Tavg25,
+    xmax = Tavg75,
+    ymin = 365 * Pr25,
+    ymax = 365 * Pr75
+  )
+) + geom_point(aes(shape = RCP, color = CF), size = 4) +
+  #PlotTheme +
+  labs(
+    title = paste(park, "- Changes in climate means in", 2050 , "by GCM run"),
+    x = "Change in annual average temperature (F)",
+    y = "Change in average annual precipitation (in)"
+  ) +
+  scale_colour_manual(values = c("gray", "darkred", "darkblue", "pink", "lightblue")) +
+  guides(color = guide_legend(title = "Climate Future")) +
+  
+  #geom_point(colour="black",size=4) +
+  theme(
+    axis.text = element_text(size = 16),
+    axis.title.x = element_text(size = 16, vjust = -0.2),
+    axis.title.y = element_text(size = 16, vjust = 0.2),
+    plot.title = element_text(
+      size = 20,
+      face = "bold",
+      vjust = 2,
+      hjust = 0.5
+    ),
+    legend.text = element_text(size = 16),
+    legend.title = element_text(size = 16)
+  ) +
+  geom_rect(color = "black", alpha = 0) +
+  geom_hline(aes(yintercept = 365 * mean(delta_pr)), linetype = 2) +
+  geom_vline(aes(xintercept = mean(delta_tavg)), linetype = 2) +
+  # highlight high/low runoff models
+  geom_point(
+    data = filter(
+      future_means,
+      GCM %in% c("CSIRO-Mk3-6-0.rcp45", "MIROC-ESM-CHEM.rcp85")
+    ),
+    aes(delta_tavg, 365 * delta_pr),
+    shape = 1,
+    color = "yellow",
+    size = 6,
+    stroke = 3
+  ) +
+  geom_mark_circle(data = filter(future_means, GCM %in% selected_gcms$GCM),
+                    aes(group = GCM), expand = 0.025, size = 1, linetype = 2) + 
+  geom_text_repel(aes(label = GCM))
+   
+
+```
+
+## Test Function
+
+```{r}
+brca_gcms <- select_GCMs("BRCA",
+           region = "SWR",
+           future_range = 2035:2065,
+           historic_range = 1979:2012,
+           low_skill_cutoff = 0.1,
+           method = "pca",
+           save = FALSE)
+```
+
+Inspect these results:
+
+```{r}
+brca_gcms %>% 
+  ggplot(aes(x = delta_tavg, y = delta_pr)) + 
+  geom_point(aes(color = CF), size = 4) +
+  geom_text_repel(aes(label=paste(centroid, GCM)))
+```
+
+# Park wide GCM selection
 
-## **Running notes for arguments that would go into a function:**
+Read in and clean list of parks and region codes (need to filter to just Conus for centroids)
 
--   park
+```{r}
+#download a list of all park names and attributes
+downloadLink <-
+  "https://irmaservices.nps.gov/datastore/v6/rest/DownloadFile/693182"
+
+#download boundary
+temp1 <- tempfile()
+download.file(downloadLink, destfile = temp1, method = "curl")
+temp2 <- tempfile()
+unzip(temp1, exdir = temp2)
+
+parks <- sf::st_read(paste0(temp2, "/Administrative_Boundaries_of_National Park_System_Units.gdb"), quiet = TRUE)
```

Comment:
could just use `get_ParkSystem()` here instead of writing the function out here

Reply:
Did not know this existed 😆 my bad! I'll swap it out.